server/src/test/java/org/elasticsearch/search/aggregations/metrics/GeoCentroidIT.java

```diff
@@ -45,6 +48,31 @@
 public class GeoCentroidIT extends AbstractGeoTestCase {
     private static final String aggName = "geoCentroid";
 
+    public void test7xIndexOnly() {
+        SearchPhaseExecutionException exception = expectThrows(SearchPhaseExecutionException.class,
+            () -> client().prepareSearch(IDX_NAME_7x) .addAggregation(geoCentroid(aggName).field(SINGLE_VALUED_GEOSHAPE_FIELD_NAME))
+            .get());
+        assertNotNull(exception.getRootCause());
+        assertThat(exception.getRootCause().getMessage(),
+            equalTo("Can't load fielddata on [geoshape_value] because fielddata is unsupported on fields of type [geo_shape]." +
+                " Use doc values instead."));
+    }
+
+    public void test7xIndexWith8Index() {
+        SearchResponse response = client().prepareSearch(IDX_NAME_7x, IDX_NAME)
```

Comment:
Not sure if am reading this correctly. If you mix index with different versions then we do not fail? Is this the expected behavior?

Reply:
I think this is intentional? We fail shards that do not have doc-values, but since the other index is valid, the aggregation result is consistent.

It errors out if queried against an index with no geo_shape field at all.

I will double check other aggregations to see what is the most consistent behavior. I assume it is the same as is shown here

Reply:
See, and this assertion checks that some shards were not successful:

```
  assertThat(response.getSuccessfulShards(), lessThan(response.getTotalShards()));
```